test/jb_splits_store/set.test.js

```diff
@@ -0,0 +1,278 @@
+    const { projectOwner, addrs, jbSplitsStore, splits } = await setup();
+
+    // Set one locked split
+    splits[1].lockedUntil = Date.now() + ONE_DAY;
+    let tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // Try to set new ones, with lock extension
+    let newSplits = makeSplits(addrs[5].address);
+
+    // New lockedUntil = old lockedUntil + 3600 sec
+    let newLockedTimestamp = Date.now() + 2 * (ONE_DAY);
+    newSplits[1].lockedUntil = newLockedTimestamp;
+    newSplits[1].beneficiary = addrs[0].address;
+
+    tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      newSplits
+    );
+
+    // Get the splits[] curently stored
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    expect(splitsStored[1].lockedUntil).to.equal(newLockedTimestamp);
+  })
+
+  it('One split with a percent at 0: revert', async function () {
+    const { projectOwner, jbSplitsStore, splits } = await setup();
+
+    // Set one 0% split
+    splits[1].percent = 0;
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        splits)
+    ).to.be.revertedWith('0x10: BAD_SPLIT_PERCENT');
+  })
+
+  it('Allocator and beneficiary both equal to address(0): revert', async function () {
+    const { projectOwner, jbSplitsStore, splits } = await setup();
+
+    // Set both allocator and beneficiary as address(0)
+    splits[1].beneficiary = ethers.constants.AddressZero;
+    splits[1].allocator = ethers.constants.AddressZero;
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        splits)
+    ).to.be.revertedWith('0x11: ZERO_ADDRESS');
+  })
+
+  it('Sum of the splits in percent > 10000000: revert', async function () {
+    const { projectOwner, jbSplitsStore, splits } = await setup();
+
+    // Set sum at 10000001
+    splits[0].percent += 1;
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        splits)
+    ).to.be.revertedWith('0x12: BAD_TOTAL_PERCENT');
+  })
+
+  it('Not project owner but has permission', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits, mockJbOperatorStore, mockJbProjects, mockJbDirectory } = await setup();
+
+    let caller = addrs[0];
+
+    // Overriding the default permission from setup()
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(caller.address, projectOwner.address, PROJECT_ID, SET_SPLITS_PERMISSION_INDEX)
+      .returns(true);
+
+    await expect(jbSplitsStore.connect(caller).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    )).to.be.not.reverted;
+  })
+
+  it('Not project owner and has no permission: revert', async function () {
```

Comment:
Not a big deal but this name doesn't match the style of the JBDirectory tests.

This should be more like: 

`Cant set splits if not project owner and doesn't have permission`